gate-oauth2/src/main/java/com/netflix/spinnaker/gate/security/oauth2/SpinnakerOIDCUser.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2025 OpsMx, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.gate.security.oauth2;
+
+import com.netflix.spinnaker.security.User;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.oauth2.core.oidc.OidcIdToken;
+import org.springframework.security.oauth2.core.oidc.OidcUserInfo;
+import org.springframework.security.oauth2.core.oidc.user.OidcUser;
+
+public class SpinnakerOIDCUser extends User implements OidcUser {
+  private Map<String, Object> attribute = new HashMap<>();
```

Comment:
attribute and authorities at least need javadoc too.  As it is, they could be final, which I think is an improvement even with no other changes.  However...

There's no mutator for them, but apparently the expected pattern is for callers to get them, and then mutate them.  Is there a way to organize things so the constructor has additional arguments for attribute and authorities...and then they could be immutable.

Same for claims I suppose, though I don't see any code that uses claims.

Reply:
https://github.com/spinnaker/gate/pull/1887/commits/af1c8db074110983aa80a588155e4b81c68de872